code/modules/mob/living/life.dm

```diff
@@ -230,14 +230,32 @@
 
 	if(machine)
 		if(!machine.check_eye(src))
-			reset_view(null)
+			reset_perspective(null)
```

Comment:
when the time comes, this needs to be `machine.check_eye(src)` and all the machines that alter perspective need to use this new system in this PR.